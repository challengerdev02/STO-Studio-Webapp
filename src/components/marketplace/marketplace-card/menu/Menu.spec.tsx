import {
  cleanup,
  fireEvent,
  queryAllByAttribute,
  queryByAttribute,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { DropDownMenu } from './index';

describe('Container', function () {
  afterEach(cleanup);
  let props = {
    menuItems: ['ABC', 'b'],
    onDropDown: jest.fn(),
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<DropDownMenu {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have DropDownMenu "This is a test DropDownMenu"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByAttribute(
        'class',
        document.body,
        'ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger'
      )[0]
    ).toBeInTheDocument();
  });

  it('should fire onDropDown callback when click', async () => {
    // @ts-ignore
    const { container } = setup({ visible: true });

    await waitFor(() => {
      const btn = queryByAttribute(
        'class',
        document.body,
        'ant-btn ant-btn-default'
      );

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(props.onDropDown).toHaveBeenCalled();
    });
  });
});
