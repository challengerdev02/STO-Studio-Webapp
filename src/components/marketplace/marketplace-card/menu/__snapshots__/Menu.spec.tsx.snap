// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match with snapshot 1`] = `
<div>
  <div
    class="ant-btn-group ant-dropdown-button"
  >
    <button
      class="ant-btn ant-btn-default"
      type="button"
    />
    <button
      class="ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger"
      type="button"
    >
      <span
        aria-label="ellipsis"
        class="anticon anticon-ellipsis"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="ellipsis"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
